mkt/ratings/views.py

```diff
@@ -27,25 +26,18 @@
 log = commonware.log.getLogger('z.api')
 
 
-class RatingPaginator(Paginator):
-    # FIXME: This is only right when ?app= filtering is applied, if no
-    # filtering or if ?user= filtering is done, it's completely wrong.
-    @property
-    def count(self):
-        try:
-            r = self.object_list[0]
-        except IndexError:
-            return 0
-        return r.addon.total_reviews
-
-
 class RatingViewSet(CORSMixin, MarketplaceView, ModelViewSet):
     # Unfortunately, the model class name for ratings is "Review".
-    queryset = Review.objects.valid().all()
+    # We prefetch 'version' because it's often going to be similar, and select
+    # related 'user' to avoid extra queries.
+    queryset = (Review.objects.valid()
+                .prefetch_related('version')
+                .select_related('user'))
```

Comment:
`.prefetch_related('version')` saves us 2 queries per review, assuming the same version is going to appear multiple times.
`.select_related('user')` does a `JOIN` and saves us one query per review.